timemachine/md/minimizer.py

```diff
@@ -140,85 +142,178 @@ def minimize_host_4d(
+        Number of steps to run MD with
+
+    pressure: float
+        Pressure to run barostat with
+
+    temperature: float
+        Temperature to run MD with
+
+    barostat_interval: integer
+        How often to run the barostat
+
+    seed: integer
+        Seed for barostat and integrator
 
     Returns
     -------
-    np.ndarray
-        This returns minimized host_coords.
+    2-tuple of host coordinates and box
+        the minimized host_coords and the new box
 
     """
     box = host_config.box
     assert box.shape == (3, 3)
 
+    minimized_host_coords = fire_minimize_host(
+        mols,
+        host_config,
+        ff,
+        mol_coords=mol_coords,
+        n_windows=minimizer_windows,
+        n_steps_per_window=minimizer_steps_per_window,
+    )
+
     host_bps, host_masses = openmm_deserializer.deserialize_system(host_config.omm_system, cutoff=1.2)
 
     num_host_atoms = host_config.conf.shape[0]
 
     if len(mols) == 1:
         top = topology.BaseTopology(mols[0], ff)
     elif len(mols) == 2:
-        top = topology.DualTopologyMinimization(mols[0], mols[1], ff)
+        top = topology.DualTopology(mols[0], mols[1], ff)
     else:
         raise ValueError("mols must be length 1 or 2")
 
     mass_list = [np.array(host_masses)]
-    conf_list = [np.array(host_config.conf)]
+    conf_list = [minimized_host_coords]
     for mol in mols:
-        # mass increase is to keep the ligand fixed
-        mass_list.append(get_mol_masses(mol) * 100000)
+        # Set ligand masses to inf to ensure ligands don't move
+        mass_list.append(np.ones(mol.GetNumAtoms()) * np.inf)
 
-    if mol_coords is not None:
-        for mc in mol_coords:
-            conf_list.append(mc)
-    else:
-        for mol in mols:
-            conf_list.append(get_romol_conf(mol))
+    if mol_coords is None:
+        mol_coords = [get_romol_conf(mol) for mol in mols]
+    for mc in mol_coords:
+        conf_list.append(mc)
 
     combined_masses = np.concatenate(mass_list)
     combined_coords = np.concatenate(conf_list)
 
     hgt = topology.HostGuestTopology(host_bps, top, host_config.num_water_atoms)
 
-    # this value doesn't matter since we will turn off the noise.
-    seed = 0
-
-    intg = LangevinIntegrator(0.0, 1.5e-3, 1.0, combined_masses, seed).impl()
+    intg = LangevinIntegrator(DEFAULT_TEMP, 1.5e-3, 1.0, combined_masses, seed).impl()
 
     x0 = combined_coords
     v0 = np.zeros_like(x0)
 
-    potentials, params = parameterize_system(hgt, ff, 1.0)
+    num_host_atoms = minimized_host_coords.shape[0]
+
+    potentials, params = parameterize_system(hgt, ff, 0.0)
+    bond_pot = next(pot for pot in potentials if isinstance(pot, HarmonicBond))
+    group_idxs = get_group_indices(get_bond_list(bond_pot), x0.shape[0])
+    # Only adjust the non-ligand group idxs, we don't want to move the ligand coordinates at all
+    non_ligand_group_idxs = [group for group in group_idxs if np.all(group < num_host_atoms)]
```

Comment:
Run the barostat with the ligands excluded from mols can be adjusted. Going to be less efficient, but does still reduce the box size.

Reply:
Correct

Reply:
Makes sense. May be worth emphasizing in the doc str.

Reply:
Emphasized in https://github.com/proteneer/timemachine/pull/1339/commits/ebec3f0a3b723a58d1b48eca0307c7761971e60b